emloop/cli/train.py

```diff
@@ -22,7 +22,10 @@ def train(config_path: str, cl_arguments: Iterable[str], output_root: str) -> No
         config = load_config(config_file=config_path, additional_args=cl_arguments)
         validate_config(config)
         logging.debug('\tLoaded config: %s', config)
-        main_loop = create_main_loop(config, output_root)
-        main_loop.run_training()
+        emloop_training = create_emloop_training(config, output_root)
+        if delete_dir:
+            print_delete_warning()
+        emloop_training.main_loop.run_training()
+        clean_output_dir(emloop_training.output_dir, delete_dir)
```

Comment:
actually, we should move this to a finally block so that we make sure it is executed (just make sure that `rmtree` does not raise an exception if the dir does not exist)